### ISSUE TITLE
Error reading certain Netlib problems

### ISSUE THREAD

**Issue by dpo** [JuliaSmoothOptimizers/QPSReader.jl#26]
```julia
julia> using QPSReader
julia> netlib_path = fetch_netlib()

julia> blend = readqps(joinpath(netlib_path, "BLEND.SIF"))
[ Info: Using 'BLEND' as NAME (l. 5)
[ Info: Using 'C' as objective (l. 98)
[ Info: Using '65' as RHS (l. 376)
ERROR: Unknown row 23.26.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] read_rhs_line!(::QPSData, ::QPSReader.MPSCard{QPSReader.FreeMPS}) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:498
 [3] readqps(::String; mpsformat::Symbol) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:800
 [4] readqps(::String) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:703
 [5] top-level scope at REPL[5]:1
 [6] eval(::Module, ::Any) at ./boot.jl:331
 [7] eval_user_input(::Any, ::REPL.REPLBackend) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
 [8] run_backend(::REPL.REPLBackend) at /Users/dpo/.julia/packages/Revise/Pcs5V/src/Revise.jl:1073
 [9] top-level scope at none:0

julia> prob = readqps(joinpath(netlib_path, "DFL001.SIF"))
[ Info: Using 'DFL001' as NAME (l. 5)
[ Info: Using 'NIL' as objective (l. 6098)
[ Info: Using 'R0006' as RHS (l. 29101)
ERROR: ArgumentError: cannot parse "R0009" as Float64
Stacktrace:
 [1] _parse_failure(::Type{T} where T, ::String, ::Int64, ::Int64) at ./parse.jl:370 (repeats 2 times)
 [2] #tryparse_internal#347 at ./parse.jl:366 [inlined]
 [3] tryparse_internal at ./parse.jl:364 [inlined]
 [4] #parse#348 at ./parse.jl:376 [inlined]
 [5] parse at ./parse.jl:376 [inlined]
 [6] read_rhs_line!(::QPSData, ::QPSReader.MPSCard{QPSReader.FreeMPS}) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:478
 [7] readqps(::String; mpsformat::Symbol) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:800
 [8] readqps(::String) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:703
 [9] top-level scope at REPL[6]:1
 [10] eval(::Module, ::Any) at ./boot.jl:331
 [11] eval_user_input(::Any, ::REPL.REPLBackend) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
 [12] run_backend(::REPL.REPLBackend) at /Users/dpo/.julia/packages/Revise/Pcs5V/src/Revise.jl:1073
 [13] top-level scope at none:0

julia> prob = readqps(joinpath(netlib_path, "FORPLAN.SIF"))
[ Info: Using 'FORPLAN' as NAME (l. 5)
[ Info: Using 'OB1PNW20' as objective (l. 21)
ERROR: Duplicate row name DEDO3 at line 23
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] read_rows_line!(::QPSData, ::QPSReader.MPSCard{QPSReader.FreeMPS}) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:363
 [3] readqps(::String; mpsformat::Symbol) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:796
 [4] readqps(::String) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:703
 [5] top-level scope at REPL[7]:1
 [6] eval(::Module, ::Any) at ./boot.jl:331
 [7] eval_user_input(::Any, ::REPL.REPLBackend) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
 [8] run_backend(::REPL.REPLBackend) at /Users/dpo/.julia/packages/Revise/Pcs5V/src/Revise.jl:1073
 [9] top-level scope at none:0

julia> prob = readqps(joinpath(netlib_path, "GFRD-PNC.SIF"))
[ Info: Using 'GFRD-PNC' as NAME (l. 5)
[ Info: Using 'OBJ' as objective (l. 20)
[ Info: Using 'PAF' as RHS (l. 2821)
ERROR: ArgumentError: cannot parse "PEE" as Float64
Stacktrace:
 [1] _parse_failure(::Type{T} where T, ::String, ::Int64, ::Int64) at ./parse.jl:370 (repeats 2 times)
 [2] #tryparse_internal#347 at ./parse.jl:366 [inlined]
 [3] tryparse_internal at ./parse.jl:364 [inlined]
 [4] #parse#348 at ./parse.jl:376 [inlined]
 [5] parse at ./parse.jl:376 [inlined]
 [6] read_rhs_line!(::QPSData, ::QPSReader.MPSCard{QPSReader.FreeMPS}) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:478
 [7] readqps(::String; mpsformat::Symbol) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:800
 [8] readqps(::String) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:703
 [9] top-level scope at REPL[8]:1
 [10] eval(::Module, ::Any) at ./boot.jl:331
 [11] eval_user_input(::Any, ::REPL.REPLBackend) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
 [12] run_backend(::REPL.REPLBackend) at /Users/dpo/.julia/packages/Revise/Pcs5V/src/Revise.jl:1073
 [13] top-level scope at none:0

julia> prob = readqps(joinpath(netlib_path, "SIERRA.SIF"))
[ Info: Using 'SIERRA' as NAME (l. 5)
[ Info: Using 'OBJ' as objective (l. 1247)
[ Info: Using 'BB1R' as RHS (l. 6566)
ERROR: ArgumentError: cannot parse "BB2R" as Float64
Stacktrace:
 [1] _parse_failure(::Type{T} where T, ::String, ::Int64, ::Int64) at ./parse.jl:370 (repeats 2 times)
 [2] #tryparse_internal#347 at ./parse.jl:366 [inlined]
 [3] tryparse_internal at ./parse.jl:364 [inlined]
 [4] #parse#348 at ./parse.jl:376 [inlined]
 [5] parse at ./parse.jl:376 [inlined]
 [6] read_rhs_line!(::QPSData, ::QPSReader.MPSCard{QPSReader.FreeMPS}) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:478
 [7] readqps(::String; mpsformat::Symbol) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:800
 [8] readqps(::String) at /Users/dpo/dev/julia/JSO/QPSReader.jl/src/readqps.jl:703
 [9] top-level scope at REPL[9]:1
 [10] eval(::Module, ::Any) at ./boot.jl:331
 [11] eval_user_input(::Any, ::REPL.REPLBackend) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
 [12] run_backend(::REPL.REPLBackend) at /Users/dpo/.julia/packages/Revise/Pcs5V/src/Revise.jl:1073
 [13] top-level scope at none:0
```

cc @geoffroyleconte

**Comment by mtanneau** [comment]
The reader errors because these files are in fixed MPS format, but Free MPS is used by default.

Notably, these files have row/column names that contain spaces (including IIRC some fieldnames that are just empty).

You need to pass `mpsformat = :fixed` to force the use of Fixed MPS:
```julia
using QPSReader
netlib_path = fetch_netlib()
blend = readqps(joinpath(netlib_path, "BLEND.SIF"), mpsformat=:fixed)
```

**Comment by dpo** [comment]
Ah, I see. Thank you. Maybe the reader could print a message in case of error saying something like "try reading the file with `mpsformat=:fixed`".

**Comment by mtanneau** [comment]
> print a message in case of error

I am not 100% in favour of doing that, mainly because there is virtually no way of telling whether an error is due to fixed vs free format, or something else. For instance, in the above example, out of 5 instances, three different errors were encountered.
For what it's worth,  Mosek and CPLEX also error when reading those files.

We should however add a line in the README (e.g., in "Usage") to show how one can select the MPS format, and maybe underline the difference between fixed & free format.

**Comment by dpo** [comment]
Sounds good.

**Comment by mtanneau** [comment]
Closed by #27
